### PR TITLE
MINOR: Remove some unused code

### PR DESCRIPTION
**Pull request by dengziming**
*More detailed description of your change*
Remove 2 classes no longer used.

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by Kvicii — APPROVED** [review]
looks great.

**Comment by dengziming** [comment]
Hello @showuon  , PTAL, Thanks.

File: core/src/main/scala/kafka/common/Config.scala [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-
-package kafka.common
-
-import util.matching.Regex
-import kafka.utils.Logging
-import org.apache.kafka.common.errors.InvalidConfigurationException
-
-trait Config extends Logging {
-
-  def validateChars(prop: String, value: String): Unit = {
```
Comment (showuon) [inline core/src/main/scala/kafka/common/Config.scala:26]: I confirmed the `validateChars` was used in old clients, and not used now anymore. It makes sense to remove it.

File: core/src/main/scala/kafka/common/BaseEnum.scala [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
- * limitations under the License.
- */
-package kafka.common
-
-/*
- * We inherit from `Product` and `Serializable` because `case` objects and classes inherit from them and if we don't
- * do it here, the compiler will infer types that unexpectedly include `Product` and `Serializable`, see
- * http://underscore.io/blog/posts/2015/06/04/more-on-sealed.html for more information.
- */
-trait BaseEnum extends Product with Serializable {
```
Comment (showuon) [inline core/src/main/scala/kafka/common/BaseEnum.scala:24]: I confirmed `BaseEnum` was used in old clients and not used now anymore.

**Review by showuon — APPROVED** [review]
LGTM! Thanks.

**Comment by showuon** [comment]
@dengziming , could you rebase to the latest trunk to trigger the jenkins build again? Thanks.

**Comment by dengziming** [comment]
> @dengziming , could you rebase to the latest trunk to trigger the jenkins build again? Thanks.
Done!

**Comment by showuon** [comment]
The latest jenkins build has no failed tests.
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11935/6/

**Pull request merged**